From d413fe3b37c3cefe19b964044403a34d074b471c Mon Sep 17 00:00:00 2001
From: Bin Jiang <bin.jiang@windriver.com>
Date: Thu, 2 May 2013 16:56:01 +0800
Subject: [PATCH] Revert "MIPS: Octeon: Make -std and -cgl kernel have the
 same context"

Don't recover staging octeon ethernet, since it has been removed.

Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 arch/mips/include/asm/thread_info.h |    7 ++++++-
 drivers/char/ipmi/ipmi_msghandler.c |    2 +-
 include/linux/rio.h                 |    2 +-
 3 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/arch/mips/include/asm/thread_info.h b/arch/mips/include/asm/thread_info.h
index dbc0e2b..8826c97 100644
--- a/arch/mips/include/asm/thread_info.h
+++ b/arch/mips/include/asm/thread_info.h
@@ -125,6 +125,8 @@ register struct thread_info *__current_thread_info __asm__("$28");
 #define TIF_LOAD_WATCH		25	/* If set, load watch registers */
 #define TIF_XKPHYS_MEM_EN	26
 #define TIF_XKPHYS_IO_EN	27
+/* li takes a 32bit immediate */
+#define TIF_GRSEC_SETXID	29	/* update credentials on syscall entry/exit */
 #define TIF_KERNEL_TRACE	30	/* kernel trace active */
 #define TIF_SYSCALL_TRACE	31	/* syscall trace active */
 
@@ -152,9 +154,12 @@ register struct thread_info *__current_thread_info __asm__("$28");
 #define _TIF_LOAD_WATCH		(1<<TIF_LOAD_WATCH)
 #define _TIF_XKPHYS_MEM_EN	(1<<TIF_XKPHYS_MEM_EN)
 #define _TIF_XKPHYS_IO_EN	(1<<TIF_XKPHYS_IO_EN)
+#define _TIF_GRSEC_SETXID	(1<<TIF_GRSEC_SETXID)
+
+#define _TIF_SYSCALL_WORK	(_TIF_SYSCALL_TRACE | _TIF_SYSCALL_AUDIT | _TIF_GRSEC_SETXID)
 
 /* work to do in syscall_trace_leave() */
-#define _TIF_WORK_SYSCALL_EXIT	(_TIF_SYSCALL_TRACE | _TIF_SYSCALL_AUDIT)
+#define _TIF_WORK_SYSCALL_EXIT	(_TIF_SYSCALL_TRACE | _TIF_SYSCALL_AUDIT | _TIF_GRSEC_SETXID)
 
 /* work to do on interrupt/exception return */
 #define _TIF_WORK_MASK		(0x0000ffef &				\
diff --git a/drivers/char/ipmi/ipmi_msghandler.c b/drivers/char/ipmi/ipmi_msghandler.c
index 15980d6..e9ad39d9 100644
--- a/drivers/char/ipmi/ipmi_msghandler.c
+++ b/drivers/char/ipmi/ipmi_msghandler.c
@@ -2898,7 +2898,7 @@ int ipmi_register_smi(struct ipmi_smi_handlers *handlers,
 	INIT_LIST_HEAD(&intf->cmd_rcvrs);
 	init_waitqueue_head(&intf->waitq);
 	for (i = 0; i < IPMI_NUM_STATS; i++)
-		atomic_set(&intf->stats[i], 0);
+		atomic_set_unchecked(&intf->stats[i], 0);
 	intf->all_cmd_rcvr = NULL;
 
 	intf->proc_dir = NULL;
diff --git a/include/linux/rio.h b/include/linux/rio.h
index b288b65..bebcdc9 100644
--- a/include/linux/rio.h
+++ b/include/linux/rio.h
@@ -327,7 +327,7 @@ struct rio_ops {
 			u64 offset, u64 length);
 	void (*unmap) (struct rio_mport *mport, struct rio_dev *rdev,
 			u64 offset, u64 length, phys_t map);
-};
+} __no_const;
 
 #define RIO_RESOURCE_MEM	0x00000100
 #define RIO_RESOURCE_DOORBELL	0x00000200
-- 
1.7.10.4

